editor/src/clj/editor/gui.clj

```diff
@@ -764,19 +1039,25 @@
                                                (not= node-id (g/node-value node-id :parent)))))
                      :children node-outline-children
                      :outline-error? (g/error-fatal? own-build-errors)
-                     :outline-overridden? (not (empty? _overridden-properties))}
+                     :outline-overridden? (if (str/blank? current-layout)
+                                            (< 1 (count _overridden-properties)) ; :layout->prop->override will always be present, and we shouldn't count it.
+                                            (pos? (count (layout->prop->override current-layout))))}
                     (resource/openable-resource? node-outline-link) (assoc :link node-outline-link :outline-reference? true))))
 
   (output transform-properties g/Any scene/produce-scalable-transform-properties)
   (output gui-base-node-msg g/Any produce-gui-base-node-msg)
   (output node-msg g/Any :abstract)
   (input node-msgs g/Any :array)
-  (output node-msgs g/Any (g/fnk [node-msgs node-msg]
-                            (into [node-msg]
-                                  (map #(dissoc % :child-index))
-                                  (flatten
-                                    (sort-by #(get-in % [0 :child-index])
-                                             node-msgs)))))
+  (output node-msgs g/Any (g/fnk [layout->prop->override layout->prop->value node-msg node-msgs]
+                            (let [decorated-node-msg
+                                  (assoc node-msg
+                                    :layout->prop->override layout->prop->override
+                                    :layout->prop->value layout->prop->value)]
```

Comment:
We decorate the `node-msgs` with these. They are used in the `build-targets` and `save-value` outputs of `GuiSceneNode`.